custom_components/modernforms/__init__.py

```diff
@@ -1,36 +1,42 @@
-import requests
-
 from datetime import timedelta
 from homeassistant.const import (CONF_SCAN_INTERVAL)
 from homeassistant.helpers.entity import Entity
-from homeassistant.helpers.event import (async_call_later, async_track_time_interval)
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.helpers.update_coordinator import (CoordinatorEntity, DataUpdateCoordinator , UpdateFailed)
+from types import MethodType
+from typing import Any
 import logging
+import httpx
 
-from .const import DOMAIN, DEVICES, CONF_FAN_NAME, CONF_FAN_HOST, CONF_ENABLE_LIGHT
+from .const import DOMAIN, DEVICES, COORDINATORS, CONF_FAN_NAME, CONF_FAN_HOST, CONF_ENABLE_LIGHT
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_LIGHT = "light"
-SCAN_INTERVAL = 60
+SCAN_INTERVAL = timedelta(seconds=5)
```

Comment:
Please see previous commit. Can you please leave this to be just a value.

Reply:
I am not sure this would actually solve it. In the past, as part of the troubleshooting from HA forum, I built a completely [standalone prober](https://github.com/icemarkom/mffprober) for the fans, and simply running this for under a day would freeze them such they required a powercycle. What your solution **_may_** help with is Lovelace locking up when fans are slow to respond and/or die. My concern is an apparent memory leak on the fans themselves. Hence recommendation to leave this as a default to 60, **and** have a configuration parameter so those who need/want faster polling can adjust (see Issue #8, and potentially Issue #15).

Reply:
I would think not all fans have the same issue. My ones (AXIS) never need restarting. 
One is on firmware 01.03.0021/01.03.3014 and the other on 01.03.0021/01.03.3008. I think the latest is 0027 but I not sure I should attempt to update them.

@faceprint, do your fans need restarting before your fix? Either way, 5 seconds is WAY to frequent. I would think 30s or 60 is better, especially for a default value

Reply:
Yes, my fans were hanging requiring a power cycle with the "stock" integration about a week ago.  I noticed all the socket connections stuck in `TIME_WAIT` state (I think I saw 64), figuring something wasn't getting cleaned up.  I had the interval set to 10 seconds, and all was well, usually having 4 or 5 connections, all but one in `TIME_WAIT`.  I decided to give 5 seconds a try, and no longer have any connections stuck in `TIME_WAIT`  and the fan is operating normally.  This is a recent discovery, so it hasn't had time to "soak" but I'm optimistic based on the behavior of the connections.  Seems like the fan "likes" seeing activity on the connection.

As long as its configurable, I'm happy with whatever default people want.

Reply:
Interesting. What firmware are you on?
If you set the interval to 5 seconds (without using httpx) does your theory still stand?

Reply:
Ohhh this is indeed an interesting find! So it may not be the polling interval, but the number of sockets that kills the fan. You know what - this is something that makes me optimistic. 

A good way to test this hypothesis would be to run that prober of mine until it kills the fan, and then see if there are any open sockets. I am almost willing to bet we'll see a nice "round" number like 64.

Then, we can try the same using HTTP/1.1 with a single persistent session. This has some drawbacks, but I am almost certain Home Assistant is smart enough to re-establish it should it disconnect. 

This is rock-solid clue, first in a long time.

Reply:
Of course, a separate issue entirely is that, right now, we don't actually have a way to override scan_interval :-).

Reply:
@icemarkom will you have time to investigate this at all? If so, we'll wait before we merge these changes

Reply:
I saw some instability at 5 seconds overnight.  No hard lockups, just a lot of timeouts, and they resolved themselves (this was all overnight), so it may have been some wifi hiccup.  I think we can merge as is with 60 seconds, and experiment with other intervals to find if there's a better sweet spot.

I don't think that using the current setup at 5 seconds without async/httpx is a useful experiment, since (I believe) its the hung connections, and not the interval between them that causes the hard lockups, and the "stock" integration seems to leave a lot of those hung connections.

Reply:
I will also test it out, but I am primary oncall in my day job until Thursday, which leaves limited time to play with the cool stuff I want to play with :-). If I find some time, will likely pull and start testing this today and let it soak for a few days.

Starting off with 10-15 seconds as the timeout may be a good thing. 5 seconds strikes me as overly excessive polling for a fan, honestly. Just sounds too chatty...

Reply:
Running for a few days with 60 seconds, saw 23 `ESTABLISHED` connections, and intermittent connectivity to the fan (even just ping has high packet loss).  Power-cycling the fan has restored normal operation for the moment.  At 60 seconds I'm (typically) seeing 1 `ESTABLISHED` and 1 `TIME_WAIT` connection at any given time.  I'm gonna try a "soak" at 10 seconds and see how that fares.

Reply:
Could this be as simple as telling the HTTP lib no to send a connection: keep-alive? forcing polls to always close the TCP connection

Reply:
I have only now gotten to the point I can start running Nathan's HTTPX integration.

Reply:
And I think I am seeing the same thing as Nathan mentioned. With 60 second timeout, there always appears to be one `ESTABLISHED` and one `TIME_WAIT` connection:

```
bash-5.0# netstat -an | grep 15[12]
tcp        0      0 172.19.0.2:33606        192.168.xx.152:80       TIME_WAIT
tcp        0      0 172.19.0.2:48762        192.168.xx.151:80       ESTABLISHED
tcp        0      0 172.19.0.2:34008        192.168.xx.152:80       ESTABLISHED
tcp        0      0 172.19.0.2:48360        192.168.7xx.151:80       TIME_WAIT
```

The connections seem to be closing and reopening. I suspect that over time they just get stuck. BUT, `httpx` thing seems to be working, and is probably the right direction. 

Let's move the codebase to it, and continue troubleshooting this in issue #15.